### DISCUSSION TITLE
Query by PK before a vector index is created?

### DISCUSSION THREAD

**Discussion by krashdifferent** [milvus-io/milvus#25216] [Q&A and General discussion]
As far as I can tell, in 2.2 a vector index is required before a collection can be loaded and any query performed. Maybe I'm missing something obvious, but is there any way to query data by primary key before creating a vector index? It would be helpful during debugging and setup if I could verify and query data before running the index creation, which can take a while.

**Comment by yhmo** [comment]
The query() interface is to retrieve data from the database: https://milvus.io/docs/query.md
Since Milvus keeps the original data in storage, you can retrieve data anytime even before/after creating a vector index.

**Reply by krashdifferent** [reply]
Thank you for the reply, however in Milvus 2.2.10, when loading a collection, I get an error if a vector index does not exist. Am I missing something else?

```python
# setup 
connections.connect('default', host=host, port=port)
schema = CollectionSchema([
    FieldSchema(name='pk', dtype=DataType.VARCHAR, is_primary=True, auto_id=False, max_length=100),
    FieldSchema(name='embedding', dtype=DataType.FLOAT_VECTOR, dim=2)
], description="test collection")
collection = Collection('query_test', schema=schema)
data = [['0', '1'], np.ones((2, 2), np.float32)]
collection.insert(data)

# load
collection.load() # error here

# query
collection.query(expr='pk >= ""', output_fields=['embedding'])
```

`RPC error: [load_collection], <MilvusException: (code=1, message=index doesn't exist, collectionID 442465100927590852)>, <Time:{'RPC start': '2023-06-29 06:20:19.011546', 'RPC error': '2023-06-29 06:20:19.056547'}>`

Without loading, an error will occur when trying to query. Separately I tried building a non-vector index, but that gives the same error as well. I can only get the collection to load and the query to work if I have built an index on a vector field.
